src/httpAuthScheme.ts

```diff
@@ -346,7 +346,11 @@ export class Token {
         }
     }
 
-    static deserialize(tokenTypeEntry: TokenTypeEntry, bytes: Uint8Array): Token {
+    protected static deserializeWithType<subClass extends Token>(
+        subClassCtor: { new (..._: ConstructorParameters<typeof Token>): subClass },
```

Comment:
write `constructor` in full letter. Also, there's no issuer passing a name to these arguments
```suggestion
        subClassConstructor: { new (...params: ConstructorParameters<typeof Token>): subClass },
```

Reply:
removed subclasses